src/_zkapauthorizer/tests/test_model.py

```diff
@@ -856,6 +865,72 @@ def test_lease_maintenance_activity(self, get_config, now, activity):
         )
 
 
+class EventStreamTests(TestCase):
+    """
+    Tests related to the event-stream storage of VoucherStore
+    """
+
+    @given(
+        tahoe_configs(),
+        posix_safe_datetimes(),
+        lists(sql_identifiers(), min_size=1),
+        tables(),
+        data(),
+        lists(sampled_from([inserts, deletes, updates]), min_size=1),
+    )
+    def test_event_stream_serialization(
+        self, get_config, now, ids, table, data, change_types
+    ):
+        """
+        Various kinds of SQL statements can be serialized into and out of
+        the event-stream.
+        """
+        # no BLOBs
+        assume(
+            not any(
+                column[1].affinity == StorageAffinity.BLOB for column in table.columns
+            )
+        )
+        tempdir = self.useFixture(TempDir())
+        store = VoucherStore.from_node_config(
+            get_config(tempdir.join("node"), "tub.port"),
+            lambda: now,
+            memory_connect,
+        )
+
+        # generate some SQL events
+        sql_statements = []
+        sequence = count(1)
+        for sql_id in ids:
+            for change_type in change_types:
+                change = data.draw(change_type(sql_id, table))
+                sql_statements.append(
+                    (next(sequence), change.bound_statement(store._connection.cursor()))
+                )
+                store.add_event(change.bound_statement(store._connection.cursor()))
+
+        events = store.get_events()
+        self.assertThat(
+            events,
+            AfterPreprocessing(
+                lambda eventstream: [
+                    (change.sequence, change.statement)
+                    for change in eventstream.changes
+                ],
+                Equals(sql_statements),
```

Comment:
Does `Change` not have a good `__eq__`?  Why unpack it like this instead of comparing the list of Changes directly?

Reply:
I suppose I create and store `Change`s in the test, yes ... i think it's unpacked because i started just comparing the statements and added sequences later.

Reply:
(The `change` from the data.draw isn't the same as `replicate.Change` ...)